core/src/main/java/org/elasticsearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -105,7 +107,7 @@ public void onNoLongerMaster(String source) {
 
                 @Override
                 public void onFailure(String source, Exception e) {
-                    logger.error("unexpected failure during [{}]", e, source);
+                    logger.error((Supplier<?>) () -> new ParameterizedMessage("unexpected failure during [{}]", source), e);
```

Comment:
The problem is that these overrides would then introduce boxing on any primitive arguments, and that is something that we want to avoid as well. I do not think it's feasible to do this.